src/org/labkey/targetedms/outliers/Outliers.java

```diff
@@ -39,16 +50,78 @@
      * @param sort: Sort object
      * @return retuns map collection
      */
-    public static TableSelector executeQuery(Container container, User user, String sql, Set<String> columnNames, Sort sort)
+    public static TableSelector executeQuery(Container container, User user, String sql, Set<String> columnNames, Sort sort, @Nullable Map<String, Object> namedParameters)
     {
         QuerySchema query = DefaultSchema.get(user, container).getSchema(TargetedMSSchema.SCHEMA_NAME);
         assert query != null;
-        return QueryService.get().selector(query, sql, columnNames, null, sort);
+        TableSelector tableSelector = QueryService.get().selector(query, sql, columnNames, null, sort);
+        if (null != namedParameters)
+        {
+            tableSelector.setNamedParameters(namedParameters);
+        }
+        return tableSelector;
     }
 
     protected static String getExclusionWhereSql(int metricId)
     {
         return " WHERE SampleFileId.ReplicateId NOT IN "
                 + "(SELECT ReplicateId FROM QCMetricExclusion WHERE MetricId IS NULL OR MetricId = " + metricId + ")";
     }
+
+    public static Map<String, Object> getNamedParametersForQCConfigurations()
+    {
+        return _namedParameters;
+    }
+
+    public static void setNamedParametersForQCConfigurations(List<QCMetricConfiguration> configurations)
+    {
+        Map<String, Object> namedParameters = new HashMap<>();
+        configurations.forEach(config -> {
+            String nameKey = METRIC_NAME + config.getId() + "$";
+            namedParameters.put(nameKey, config.getName());
+
+            if (null != config.getSeries1Label())
+            {
+                String labelKey = METRIC_LABEL_ONE + config.getId() + "$";
```

Comment:
What if we make this part of the query so that it fetches the name from the config table itself? That way the SQL would only need to include the RowId.